crowdfunding/projects/serializers.py

```diff
@@ -0,0 +1,86 @@
+from rest_framework import serializers
+# for every model, create a serializer.
+# You can use a model serializer - like a model form.
+# Can build itself off a model. Automates
+
+from .models import Project, Pledge, Comment
+
+class PledgeSerializer(serializers.ModelSerializer):
+    '''
+    SlugRelatedField:
+    changing the way project title displays in GET request. Displays the title of the project and not the id # using slug (human readable label)
+
+    '''
+    project = serializers.SlugRelatedField(
+        queryset = Project.objects.all(),
+        slug_field='title'
+    )
+    supporter = serializers.SerializerMethodField()
+    class Meta:
+        model = Pledge
+        fields = ['id','amount','comment','anonymous','project','supporter']
+        read_only_fields = ['id', 'supporter'] # added to remove the needs to input a supporter {automates to logged in user}
+
+    def get_supporter(self, instance):
+        '''
+
+        if the instance (supporter) has anonymous = True:
+            replace True with "anonymous"
+        else
+            replace False with the username of the supporter
+        '''
+        if instance.anonymous:
+            return "anonymous"
+        else:
+            return instance.supporter.username
+
+class PledgeDetailSerializer(PledgeSerializer):
+    '''
+    Slug:changing the way project title displays in POST or PUT request. Displays the title of the project and not the id # using slug (human readable label)
+    '''
+
+    project = serializers.SlugRelatedField(
+        queryset = Project.objects.all(),
+        slug_field='title'
+    )
+    class Meta:
+        model = Pledge
+        fields = ['id','amount','comment','anonymous','project','supporter']
+        read_only_fields = ['id', 'supporter','amount','project']
+
+class ProjectSerializer(serializers.ModelSerializer):
+    owner = serializers.ReadOnlyField(source='owner_id')
```

Comment:
you could probably have removed this line as you've declared it again lower down